# Teste

O paper, na verdade, é composto de vários experimentos menores. esses foram destrinchados a seguir:
**Reconstruction of biosphere-level metabolism**

1. Assembly of all known metabolic reactions into a biosphere-level (or pangenome) metabolic model using the KEGG[^1] database. 
    * Download using the [KEGG REST API](http://www.kegg.jp/kegg/docs/keggapi.html). 
    * Construct a stoichiometric matrix[^2] from the KEGG reaction database using reaction equations.
    * Remove reactions that:
      * Are elementally imbalanced for any element, except hydrogen;
      * Consume or produce compunds that (i) do not include a SMILES string[^3] or (ii) include an n-subunit polymer with undefined molecular formulas. Metabolites with arbitrary, ‘‘R’’ groups were retained as long as ‘‘R’’ groups were balanced in the reaction equations.

    Expected result: Elementally balanced network that consists of 6880 reactions and 5944 metabolites (on Tables S1A and S1B). This network is referenced as:
      * "full KEGG network"
      * "KEGG network"
    * "Aerobic Network"
  <!--
- Essa parte do experimento foi pulada pelo fato de necessitar conhecimento específico da área que eu não faço ideia como fazer. De qq jeito aprendi coisas interessantes como
Stoichometric Matrix:
Smiles String:
 Note that due to the filtering of KEGG reactions described above, essential for an accurate accounting of atoms, the KEGG networks used throughout the manuscript are depleted in enzymes that catalyze reactions that are chemically unbalanced (e.g., fatty acid elongation reactions, lysine biosynthesis), and in enzymes with no assigned KEGG reaction, like arsenate reductase (EC 1.20.4.4) and ketol-acidreductoi-somerase (EC 1.1.1.382). We expect that the accuracy of future analyses will increase with further improvement of KEGG and other
metabolic databases, which will better reflect the collection of biosphere-level metabolism.
-->
This part of the experiment was skipped because I don't have the know-how of the area to replicate it.

---
2. The results presented in the Figure 1 and supplemental figures S1 and S2 were generated by applying the network expansion algorithm to different variants of the KEGG network, in which we generally assume all reactions to be reversible (by representing each reaction twice in both forward and backward directions). The only exception is the set of reactions that involve molecular oxygen, which were not allowed to proceed in the oxygen-producing direction, in order to best mimic the conditions of the biosphere prior to the great oxidation, as explored previously (Raymond and Segrè, 2006).
    | Seed metabolites | 
    |:----------------:|
    | H~2~O |
    | CO~2~ (HCO~3~^-^) |
    | H~2~S |
    | NH~3~ (Ammonia) |
    | N~2~ |
    | Formate |
    | Acetate |
    ![The San Juan Mountains are beautiful!](assets/gr1_lrg.jpg "San Juan Mountains")
    *A single track trail outside of Albuquerque, New Mexico.*
3. 

While the above constraints on oxygen-involving reactions can be used to study the growth of metabolism through the network expansion algorithm, they cannot easily used for imposing pre-oxic conditions in enrichment analyses that use sets of KEGG reactions andenzymes. In particular, as shown in Figure 2, for establishing the uniqueness of some properties of the phosphate-free network, we performed tests of statistical significance for enrichment of these properties relative to both the full KEGG network (aerobic) and the network accessible without oxygen (anaerobic network). The comparisons with the anaerobic network were performed in order to ensure that statistical enrichment tests were not biased by including reactions and enzymes likely added to the global metabolic network after oxygen accumulated in the atmosphere. The anaerobic network was generated by removing subsets of reactions and metabolites from the global metabolic network reachable only through reactions that utilize molecular oxygen, resulting in a modified biosphere-level metabolic network. This network was obtained through the following steps: We first removed all reactions that utilize oxygen. Second,the stoichiometric matrix was converted into a bipartite undirected graph, where nodes were either reactions or metabolites. In this bipartite graph, an edge exists between a reaction and a metabolite if that reaction either consumes or produces that metabolite.
The graph was used as an input into the python package NetworkX ([https://networkx.github.io/](https://networkx.github.io/)), and all connected components were detected. In our specific analysis, this algorithm identified a single major connected component that contained the majority of metabolic reactions. The final anaerobic network, containing 5,651 reactions and 5,252 metabolites, is provided in the Tables S1A and S1B.
**essa tabela nao existe!!**

Seed metabolites
H2O
CO2 (HCO3-)
H2S
NH3 (Amonia!)
N2
Formate
Acetate

3. Reduced carbon is an essential component of metabolism, requiring either an autotrophic carbon fixation process or the hetero-
trophic carbon assimilation of abiotically reduced carbon. We tested two scenarios: (i) an autotrophic origin of metabolism from car-
bon dioxide and hydrogen gas and (ii) a heterotrophic origin of metabolism from formate and acetate. We did not see significant
growth from scenario (i), indicating that a reduced form of carbon is required. Acetate and formate were chosen based on previous
work suggesting that early forms of abiotically fixed carbon may have existed in the form of simple carboxylic acids. These acids
could have been in principle synthesized at hydrothermal vents from hydrogen and carbon dioxide using the processes of serpenti-
nization (Lang et al., 2010; Martin and Russell, 2007; Russell et al., 2010), or via a primitive variant of a modern carbon fixation
pathway such as the Wood-Ljungdahl pathway (Fuchs, 2011; Sousa and Martin, 2014; Sousa et al., 2013; Weiss et al., 2016). We
explored variations to this seed set in two ways. First, we performed a Monte Carlo permutation test on the seed set (see Figure S1)
and second, we varied the identity of the carbon sources (see Figure S2).

To identify potential thermodynamic barriers, we performed network expansion without unidirectional reactions (see Network
Expansion Algorithm section) above a predefined free energy threshold, t. In this variant of network expansion, reaction r was
0o
removed if DG0o
r > t, for varying levels of t. We obtained estimates for DGr from eQuilibrator (Flamholz et al., 2012), which uses
the component contribution method to estimate free energies of formation of metabolites based on the group decomposition of
compounds (Noor et al., 2013). We obtained estimates of DG0o
r at various pH values, ranging from pH 5 to pH 9 in increments of
0.5, while assuming a constant ionic strength of 0.1 M and temperature of 298.15 K. We performed network expansion at
thresholds varying from 0 to 60 kJ/mol in 1 kJ/mol increments. Final network sizes in all scenarios were insensitive to the choice
of pH.
Over one third of all KEGG reactions did not have estimates for DG0o
r , due to the large set of metabolites with no estimate for the free
energy of formation. We accounted for this by either assuming (i) all reactions with unknown DG0o
r were feasible regardless of the
cutoff or (ii) reactions with no estimate for DG0o
r were infeasible, and subsequently removed altogether. The qualitative results pre-
sented in Figure 3 of the main text are unaffected by the treatment of these reactions.

```
Here's a simple footnote,[^1] and here's a longer one.[^bignote]

[^1]: This is the first footnote.
```

[^1]: KEGG: Kyoto Encyclopedia of Genes and Genomes
[^2]: Stoichiometric matrix:
[^3]: SMILES String:  